cli/azd/internal/installed_by.go

```diff
@@ -0,0 +1,32 @@
+package internal
```

Comment:
I would rather we create a `pkg/installer` and put this there so at call sites it looked like `installer.GetRawInstalledBy()`.

I would also have us export something like:

InstalledBy() InstallType

Where InstallType is something like:

type InstallType string

and then we define a constants like `InstallTypeBrew`, `InstallTypeWinGet` as well as `InstallTypeUnknown`, so other components can just switch on these known values.

I would use that new InstallType method in your implementation (and keep GetRawInstalledBy() for the telemetry system which wants to hash the file).